src/Symfony/Component/Messenger/Transport/Receiver/ReceiverInterface.php

```diff
@@ -23,11 +27,31 @@ interface ReceiverInterface
      *
      * The handler will have, as argument, the received {@link \Symfony\Component\Messenger\Envelope} containing the message.
      * Note that this envelope can be `null` if the timeout to receive something has expired.
+     *
+     * If the received message cannot be decoded, the message should not
+     * be retried again (e.g. if there's a queue, it should be removed)
+     * and a MessageDecodingFailedException should be thrown.
+     *
+     * @throws TransportException If there is an issue communicating with the transport
      */
     public function receive(callable $handler): void;
 
     /**
      * Stop receiving some messages.
      */
     public function stop(): void;
+
+    /**
+     * Acknowledge that the passed message was handled.
+     *
+     * @throws TransportException If there is an issue communicating with the transport
+     */
+    public function acknowledge(Envelope $envelope): void;
```

Comment:
Suggestion: rename `acknowledge()` (which is hard to type without a typo, at least for me) to `ack()`?